### DISCUSSION TITLE
How to close drawer when class lg:drawer-open is set?

### DISCUSSION THREAD

**Discussion by muhammedmisir** [saadeghi/daisyui#2784] [Q&A] [answered]
Hi Everyone

I am implementing the drawer in the current version (4.6.0). I have the following use-case. On the desktop version i want to keep the drawer open like discribed in the documentation and want to give the user the possibility to close the drawer to get the full content. 

I have prepared a demo for this. You have to open it in the desktop view to understand the issue.

https://play.tailwindcss.com/wved1Vh7ow

How can i archieve this ?

**Comment by saadeghi** [comment, marked as answer]
What checking the checkbox does (by clicking the label) is to show an off-canvas sidebar.  
What `drawer-open` class does is to show sidebar and content side-by-side.

So if you want to have a off-canvas menu on mobile AND you want to have toggleable side-by-side layout on desktop, that's **two different functionalities**.  
You can show the `<label for="my-drawer-2" class="btn btn-primary drawer-button">` only for mobile (by adding `lg:hidden`). Then you can add _another_ button only for desktop screen (using `max-lg:hidden`) which adds/removes `lg:drawer-open` class when clicked (using JS)

Let me know if you have a question.
